bitcoin/src/bip32.rs

```diff
@@ -450,8 +449,11 @@ impl DerivationPath {
 
 impl fmt::Display for DerivationPath {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        f.write_str("m")?;
-        for cn in self.0.iter() {
+        let mut iter = self.0.iter();
+        if let Some(first_element) = iter.next() {
+            fmt::Display::fmt(first_element, f)?;
```

Comment:
Still getting used to rust so pardon my ignorance, but this shouldn't be changing anything. Previously, the code was writing `m` directly to the formatter and then calling `fmt::Display::fmt(cn, f)?` in a loop. My understanding is `fmt::Display::fmt(cn, f)` is being called so that the `ChildNumber` types get printed with the hardened notation.

Since we no longer start the strings with `m` my change is consuming the first element from the iterator (which is a `ChildNumber`) and calling `fmt::Display::fmt` so that the first element is properly displayed with the hardened notation before proceeding with the rest of the elements.

I'm probably misunderstanding your point, but AFAICT this is keeping the behavior for displaying derivation paths the same as before.

Reply:
Formatter can have options that are passed from the format string (e.g. `{:42}` pads the content to 42 chars). By passing formatter directly you keep the format options so if the user writes `{:5}` Instead of `  1/2` your code will output `    1/    2`.

My suggestion removes formatting options so they won't work at all which I think is better than having it broken. We can improve it later.

Reply:
The old code was also wrong :).

The issue is that if you try to `format!("{:20}", path)` say, the existing code (and your modification) will apply the `:20` to each of the child numbers, instead of applying it to the whole path.

Reply:
Gotcha! Thanks for the explanation @Kixunil @apoelstra , will update to use @Kixunil 's suggestion.